bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/discovery/OmnilinkDiscoveryService.java

```diff
@@ -0,0 +1,543 @@
+    private final Logger logger = LoggerFactory.getLogger(OmnilinkDiscoveryService.class);
+    private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    private OmnilinkBridgeHandler bridgeHandler;
+    private @Nullable SystemType systemType;
+    private @Nullable List<AreaProperties> areas;
+
+    private static final Set<Integer> TEMP_SENSOR_TYPES = Collections
+            .unmodifiableSet(Stream.of(AuxSensorProperties.SENSOR_TYPE_EXTENDED_RANGE_OUTDOOR_TEMPERATURE,
+                    AuxSensorProperties.SENSOR_TYPE_EXTENDED_RANGE_TEMPERATURE,
+                    AuxSensorProperties.SENSOR_TYPE_EXTENDED_RANGE_TEMPERATURE_ALARM,
+                    AuxSensorProperties.SENSOR_TYPE_OUTDOOR_TEMPERATURE, AuxSensorProperties.SENSOR_TYPE_TEMPERATURE,
+                    AuxSensorProperties.SENSOR_TYPE_TEMPERATURE_ALARM).collect(Collectors.toSet()));
+
+    /**
+     * Creates an OmnilinkDiscoveryService.
+     *
+     * @param bridgeHandler Bride to discover things on
+     */
+    public OmnilinkDiscoveryService(OmnilinkBridgeHandler bridgeHandler) {
+        super(Collections.singleton(new ThingTypeUID(BINDING_ID, "-")), DISCOVER_TIMEOUT_SECONDS, false);
+        this.bridgeHandler = bridgeHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting scan");
+        try {
+            SystemInformation systemInformation = bridgeHandler.reqSystemInformation();
+            this.systemType = SystemType.getType(systemInformation.getModel());
+            this.areas = discoverAreas();
+            discoverUnits();
+            discoverZones();
+            discoverButtons();
+            discoverThermostats();
+            discoverAudioZones();
+            discoverAudioSources();
+            discoverTempSensors();
+            discoverHumiditySensors();
+            discoverLocks();
+        } catch (OmniInvalidResponseException | OmniUnknownMessageTypeException | BridgeOfflineException e) {
+            logger.debug("Received error during discovery", e);
+        }
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    /**
+     * Calculate the area filter the a supplied area
+     *
+     * @param area Area to calculate filter for.
+     * @return Calculated Bit Filter for the supplied area. Bit 0 is area 1, bit 2 is area 2 and so on.
+     */
+    private static int bitFilterForArea(AreaProperties areaProperties) {
+        return BigInteger.ZERO.setBit(areaProperties.getNumber() - 1).intValue();
+    }
+
+    /**
+     * Discovers OmniLink buttons
+     *
+     * @throws OmniInvalidResponseException
+     * @throws OmniUnknownMessageTypeException
+     * @throws BridgeOfflineException
+     */
+    private void discoverButtons()
+            throws OmniInvalidResponseException, OmniUnknownMessageTypeException, BridgeOfflineException {
```

Comment:
These exceptions are never thrown. Same for several other method signatures in this class.